ansible_collections/arista/avd/plugins/plugin_utils/eos_designs_shared_utils/wan.py

```diff
@@ -339,3 +339,78 @@ def should_connect_to_wan_rs(self: SharedUtils, path_groups: list) -> bool:
             local_path_group["name"] in path_groups and any(wan_interface["connected_to_pathfinder"] for wan_interface in local_path_group["interfaces"])
             for local_path_group in self.wan_local_path_groups
         )
+
+    @cached_property
+    def wan_ha(self: SharedUtils) -> bool:
+        """
+        Only trigger HA if 2 devices are in the same group and ha.enabled is true
+        """
+        if self.cv_pathfinder_role in [None, "pathfinder"]:
+            return False
+        return get(self.wan_site, "ha.enabled", default=True) and len(self.switch_data_node_group_nodes) == 2
+
+    @cached_property
+    def wan_ha_path_group_name(self) -> str:
+        """
+        Return HA path group name for the WAN design.
+
+        TODO make this configurable
+        """
+        return "LAN_HA"
+
+    @cached_property
+    def wan_ha_interfaces(self: SharedUtils) -> list:
+        """
+        Return the list of WAN HA interfaces
+        For now only picking up uplink interfaces in VRF default on the router.
+        """
+        # TODO probably does not need the whole uplink
+        return [uplink for uplink in self.get_switch_fact("uplinks") if get(uplink, "vrf") is None]
+
+    @cached_property
+    def wan_ha_peer_interfaces(self: SharedUtils) -> str:
+        return self.get_wan_peer_fact("wan_ha_interfaces")
+
+    @cached_property
+    def wan_ha_peer(self: SharedUtils) -> str | None:
+        # TODO - do we want a better mechanism here
```

Comment:
Either do the checks in all these functions and return None if wan_ha is not true, alternatively assume wan_ha and add the check in the places where you call these functions (but not extra small wrappers like you have currently)